topicgenerator/src/main/java/io/dockstore/topicgenerator/client/cli/TopicGeneratorClient.java

```diff
@@ -97,16 +102,42 @@ public static void main(String[] args) {
             final TopicGeneratorConfig topicGeneratorConfig = new TopicGeneratorConfig(config);
             final TopicGeneratorClient topicGeneratorClient = new TopicGeneratorClient();
 
-            if ("generate-topics".equals(jCommander.getParsedCommand())) {
-                // Read CSV file
-                topicGeneratorClient.generateTopics(topicGeneratorConfig, generateTopicsCommand.getEntriesCsvFilePath());
-            } else if ("upload-topics".equals(jCommander.getParsedCommand())) {
-                // Read CSV file
-                topicGeneratorClient.uploadTopics(topicGeneratorConfig, uploadTopicsCommand.getAiTopicsCsvFilePath());
+            switch (jCommander.getParsedCommand()) {
+            case "get-topic-candidates" -> topicGeneratorClient.getAITopicCandidates(topicGeneratorConfig, getTopicCandidates.getEntriesCsvOutputFilePath());
+            case "generate-topics" -> topicGeneratorClient.generateTopics(topicGeneratorConfig, generateTopicsCommand.getEntriesCsvFilePath());
+            case "upload-topics" -> topicGeneratorClient.uploadTopics(topicGeneratorConfig, uploadTopicsCommand.getAiTopicsCsvFilePath());
+            default -> errorMessage("Unknown command", GENERIC_ERROR);
             }
         }
     }
 
+    public void getAITopicCandidates(TopicGeneratorConfig topicGeneratorConfig, String outputCsvFilePath) {
+        final ApiClient apiClient = setupApiClient(topicGeneratorConfig.dockstoreServerUrl(), topicGeneratorConfig.dockstoreToken());
+        final ExtendedGa4GhApi extendedGa4GhApi = new ExtendedGa4GhApi(apiClient);
+        LOG.info("Getting AI topic candidates from {}", topicGeneratorConfig.dockstoreServerUrl());
+
+        List<EntryLiteAndVersionName> aiTopicCandidates = extendedGa4GhApi.getAITopicCandidates();
+        LOG.info("There are {} AI topic candidates", aiTopicCandidates.size());
+
+        if (aiTopicCandidates.isEmpty()) {
+            LOG.info("No AI topic candidates found");
+            return;
+        }
+
+        try (CSVPrinter csvPrinter = new CSVPrinter(new FileWriter(outputCsvFilePath, StandardCharsets.UTF_8), CSVFormat.DEFAULT.builder().setHeader(GenerateTopicsCommand.InputCsvHeaders.class).build())) {
+            aiTopicCandidates.forEach(aiTopicCandidate -> {
+                try {
+                    csvPrinter.printRecord(aiTopicCandidate.getEntryLite().getTrsId(), aiTopicCandidate.getVersionName());
+                } catch (IOException e) {
+                    LOG.error("Could not write record for TRS ID {}, version {}", aiTopicCandidate.getEntryLite().getTrsId(), aiTopicCandidate.getVersionName());
```

Comment:
I wonder if it's worth tracking these. I suppose the code I'm suppose to write could capture these in CloudWatch?